ts/components/messages/MessageDetailsComponent.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from "react";
+
+import { Icon, Text, View } from "native-base";
+import { connectStyle } from "native-base-shoutem-theme";
+import mapPropsToStyleNames from "native-base/src/utils/mapPropsToStyleNames";
+import { TouchableOpacity } from "react-native";
+import I18n from "../../i18n";
+import MessageDetailsInfoComponent from "./MessageDetailsInfoComponent";
+
+export type OwnProps = Readonly<{
+  serviceOrganizationName: string;
+  serviceDepartmentName: string;
+  service: string;
+  key: string;
+  date: Date;
+  markdown: string;
+}>;
+
+type State = Readonly<{
+  isMessageDetailsInfoVisible: boolean;
+}>;
+
+export type Props = OwnProps;
+
+/**
+ * Implements a component that show the message details
+ */
+class MessageDetailsComponent extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { isMessageDetailsInfoVisible: false };
+  }
+  public handleDetailsInfoClick = (isVisible: boolean) => {
+    this.setState({ isMessageDetailsInfoVisible: !isVisible });
+  };
+  public renderDetailsInfoLink = () => {
+    if (!this.state.isMessageDetailsInfoVisible) {
+      return (
+        <React.Fragment>
+          <Text link={true}>
+            {I18n.t("messageDetails.detailsLink.showLabel")}
+          </Text>
+          <Icon name="chevron-right" />
+        </React.Fragment>
+      );
+    } else {
+      return (
+        <React.Fragment>
+          <Text link={true}>
+            {I18n.t("messageDetails.detailsLink.hideLabel")}
+          </Text>
+          <Icon name="cross" />
+        </React.Fragment>
+      );
+    }
+  };
+  public render() {
+    const {
+      markdown,
+      date,
+      serviceOrganizationName,
+      serviceDepartmentName,
+      service
+    } = this.props;
+    return (
+      <View>
+        <TouchableOpacity
+          onPress={() =>
+            this.handleDetailsInfoClick(this.state.isMessageDetailsInfoVisible)
+          }
+        >
+          <View>{this.renderDetailsInfoLink()}</View>
+        </TouchableOpacity>
+        {this.state.isMessageDetailsInfoVisible && (
+          <MessageDetailsInfoComponent
+            date={date}
+            service={service}
```

Comment:
if you pass the whole service why you also need serviceDepartmentName and serviceOrganizationName

Reply:
Service is only service type eg. Pagamento, https://projects.invisionapp.com/d/main#/console/13806972/287731268/preview

would it be more correct to call it serviceType?